shell/browser/win/select_file_dialog_factory.cc

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 Microsoft GmbH.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/win/select_file_dialog_factory.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/feature_list.h"
+#include "base/functional/bind.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/win/win_util.h"
+#include "chrome/browser/win/util_win_service.h"
+#include "chrome/services/util_win/public/mojom/util_win.mojom.h"
+#include "mojo/public/cpp/bindings/remote.h"
+#include "mojo/public/cpp/bindings/sync_call_restrictions.h"
+#include "ui/shell_dialogs/execute_select_file_win.h"
+#include "ui/shell_dialogs/select_file_dialog_win.h"
+#include "ui/shell_dialogs/select_file_policy.h"
+
+// Helper class to execute a select file operation on a utility process. It
+// hides the complexity of managing the lifetime of the connection to the
+// UtilWin service.
+class UtilWinHelper {
```

Comment:
Any customizations on top of https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/win/chrome_select_file_dialog_factory.h ? Can we reuse the upstream class ?

Reply:
@deepak1556 no, we just typically don't use `//chrome` factories from upstream historically, which is why i went this route! If we're fine with that in this instance i can make that change

Reply:
Don't we have bunch of them in https://github.com/electron/electron/blob/main/chromium_src/BUILD.gn ? The factory is just the client side of implementation, the core of work done by the utility process service implementation in https://source.chromium.org/chromium/chromium/src/+/main:chrome/services/util_win/util_win_impl.cc is still some we should import with patches or copy to avoid chrome specific stuff.

Reply:
There are some AV and shortcut related interfaces which are chrome specific in https://source.chromium.org/chromium/chromium/src/+/main:chrome/services/util_win/util_win_impl.cc that are not used in this PR. We should patch them out to avoid compilation failures.

Reply:
Seems the target builds fine without patching, I am fine including those unused interfaces but we should be cautious that this will break with upgrades and we might need to patch in the future.